8.x/core-concepts/cruds.md

```diff
@@ -118,204 +74,234 @@ class Todo extends BaseModel
+namespace App\Http\Controllers\Admin;
 
-```php
-    protected $fillable = [
-        "subject", "status"
-    ];
-```
+use App\Models\Todo;
+use Illuminate\Contracts\Foundation\Application;
+use Illuminate\Contracts\View\Factory;
+use Illuminate\Contracts\View\View;
+use Illuminate\Http\RedirectResponse;
+use Illuminate\Http\Request;
 
-#### $SORTABLE_FIELDS
----
-Using this property, it is possible to define for each field on what basis the user can sort the list of fields.
+/**
+ * @role(enabled=true)
+ */
+class TodoController extends BaseController
+{
+    
+    public function index(): Factory|View|Application
+    {
 
-```php
-    protected static array $SORTABLE_FIELDS = [
-        #Sort by id, status, creation date 
-        "id", "status", "created_at" 
-    ];
-```
+    }
+    
+    public function create(): Factory|View|Application
+    {
+        
+    }
 
-#### $SEARCHABLE_FIELDS
----
-This property is a list of fields and allows the user to search for a field in the list of fields.
+    public function store(Request $request): RedirectResponse
+    {
 
-```php
-    protected static array $SEARCHABLE_FIELDS = [
-        #Search by subject
-        "subject" 
-    ];
-```
+    }
 
-#### $EXACT_SEARCH_FIELD
----
-This property makes a column of the database table more important in the search system.
+    public function show(Todo $todo)
+    {
+        
+    }
 
-```php
-    #Increasing the importance of the subject column of the database table in the search system
-    protected static ?string $EXACT_SEARCH_FIELD = "subject"; 
-```
+    public function edit(Todo $todo): Factory|View|Application
+    {
 
-#### $EXACT_SEARCH_ORDER_FIELD
----
-This property configures what field the search results are sorted by.
+    }
 
-```php
-    #Search results are sorted based on creation date
-    protected static ?string $EXACT_SEARCH_ORDER_FIELD = "created_at";
-```
+    public function update(Request $request, Todo $todo): RedirectResponse
+    {
 
-### Step 3: Add Veiws
-At this stage, the view files should be created, but before that, let's add the route path and the icon file to the Larammerce project as follows.
+    }
 
-#### Add resource route
-Here, the source path for the to-do list CRUD should be added. Therefore, the following route should be written in the `routes/web.php` file.
+    public function destroy(Todo $todo): RedirectResponse
+    {
 
-##### EXAMPLE
-```php
-#TodoModel
-Route::resource("todo", "TodoController", ["as" => "admin"]);
-```
+    }
 
-:::tip NOTE
-The `web.php` file contains all system routes and including admin routes.
-:::
+    public function getModel(): ?string
+    {
+        return Todo::class;
+    }
+}
 
-####  Add an icon to the admin panel
-One of the most important parts of this tutorial is adding the task list icon in the management panel so that you can manage the tasks in it.
-It can be easily added to the admin panel by following a few simple steps.
+```
 
```

Comment:
The next step is to display this Todo list on the admin panel to be able to use it. Here the Todo list is going to be a main section on the admin panel toolbar. 
[ Move image 04 from line 43 to here]
then write this:
To do so follow the steps below: